### ISSUE TITLE
Attributes and Links limit may be exporter concern, not the runtime

### ISSUE THREAD

**Issue by SergeyKanzhelev** [census-instrumentation/opencensus-specs#227]
In Java implementation limits of attributes and links is a runtime concern. API wouldn't allow to create more then limit number of attributes.

However this approach will always call into issues like https://github.com/census-instrumentation/opencensus-specs/issues/226. Perhaps these limits can be a config setting of exporter.

Proto doesn't define any limit on either of those so limits seems to be arbitrary.

**Comment by bogdandrutu** [comment]
Do you suggest to remove any limitation on the runtime implementation and allow exporters to do whatever they consider?

**Comment by SergeyKanzhelev** [comment]
Yes. Is the need for runtime check is to protect from abuse of an API?

**Comment by pjanotti** [comment]
It seems better to leave that to exporters and also agent/collector that can be configured to do something intelligent about that (select, drop, trim, provide observability for data dropped etc).

**Comment by flands** [comment]
Agreed -- one of the design decisions for OC service was to set and forget client libraries, but with the current implementation that is not possible (already aware of someone who hit the limit and then had to reconfigure the client).
